### PR TITLE
fix: code quality issues

### PR DESCRIPTION
**Pull request by withshubh**
#### Description

Hi :wave: I ran the [DeepSource](https://deepsource.io) static analyzer on the forked copy of this repo and found some [interesting code quality issues](https://deepsource.io/gh/withshubh/torrent/issues/). This PR fixes a few of them.

#### Summary of changes

- Removed empty default in select
- Removed unnecessary use of slice
- Fixed check for empty string
- Added .deepsource.toml

### CODE REVIEW

File: webtorrent/tracker_client.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func (tc *TrackerClient) doWebsocket() error {
 				}
 			case <-closeChan:
 				return
-			default:
```
Comment (anacrolix) [inline webtorrent/tracker_client.go:93]: @DigitalAlchemist could you comment on this? I had staticcheck mention this too: There's a loop with a default clause here that could cause busy waiting if neither case is ready. Does it need to be fixed?

File: fs/torrentfs.go [hunk 2]
```diff
@@ -60,14 +60,14 @@ var (
 )
 
 func isSubPath(parent, child string) bool {
-	if len(parent) == 0 {
```
Comment (anacrolix) [inline fs/torrentfs.go:63]: @withshubh I need some justification for this.

**Review by anacrolix — CHANGES_REQUESTED** [review]

File: .deepsource.toml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+version = 1
```
Comment (anacrolix) [inline .deepsource.toml:1]: Is it necessary to have a turd file for this?

File: fs/torrentfs.go (hunk 2 above)
Comment (withshubh) [reply fs/torrentfs.go:63]: @anacrolix Please have a look at this issue description [here](https://deepsource.io/gh/withshubh/torrent/issue/CRT-A0004/description/).

File: .deepsource.toml (hunk 3 above)
Comment (withshubh) [reply .deepsource.toml:1]: It is the configuration file that is required to be present at the root of the repo. You just have to merge the PR with this file and then you can simply activate the analysis from the [dashboard](https://deepsource.io/gh/anacrolix/torrent/).

You can also use DeepSource to track test coverage, detect problems in Dockerfiles, etc. in addition to detecting issues in Go.

Comment (anacrolix) [reply .deepsource.toml:1]: Ok thanks.

File: fs/torrentfs.go (hunk 2 above)
Comment (anacrolix) [reply fs/torrentfs.go:63]: That seems fair.

**Review by anacrolix — COMMENTED** [review]
Just waiting on @DigitalAlchemist

**Review by DigitalAlchemist — APPROVED** [review]

File: webtorrent/tracker_client.go (hunk 1 above)
Comment (anacrolix) [reply webtorrent/tracker_client.go:93]: Merging...

**Pull request merged**